tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -2106,6 +2106,28 @@ public:
    /// ~~~
    unsigned int GetNRuns() const { return fLoopManager->GetNRuns(); }
 
+   /// \brief Gets descriptive information about the datasource
+   /// \return Info describing the datasource
+   ///
+   /// The information returned by this convenience function is meant for interactive
+   /// use. The exact string format should not be parsed automatically and can be subject to change.
+   ///
+   /// Example usage:
+   /// ~~~{.cpp}
+   /// ROOT::RDataFrame df("tree", "file.root");
+   /// std::cout << df.GetDataSourceInfo() << std::endl;
+   /// // prints "TTree datasource"
+   /// ~~~
+   std::string GetDataSourceInfo() const {
```

Comment:
I'm now thinking that "Info" is too generic, what do you think of `GetDataSourceKind`?

Reply:
`DataSourceKind` is good, but I think `GetDataFormatName` better reflects the return value of this function. I see it as being parallel to the recent change in the rdf docs https://root.cern/doc/master/classROOT_1_1RDataFrame.html#other-file-formats

Reply:
Eh, names are hard... TTrees and TChains are the same data format. "No datasource" is not really a data format.

Reply:
True, in fact I'd return "TTree" data format for both TTree and TChain. For the empty RDF with entries the semantic it's different, I agree.

Reply:
I think this method would be more useful if it distinguished between TTree and TChain ([as you also mentioned](https://github.com/root-project/root/issues/7828#issuecomment-817820448))